system/GD32F30x_firmware/GD32F30x_usbd_library/usbd/Source/usbd_lld_core.c

```diff
@@ -412,15 +412,15 @@ static void usbd_ep_stall_clear (usb_dev *udev, uint8_t ep_addr)
         udev->transc_in[ep_num].ep_stall = 0U;
 
         /* clear endpoint stall status */
-        USBD_EP_TX_STAT_SET(ep_num, EPTX_VALID);
+        USBD_EP_TX_STAT_SET(ep_num, EPTX_NAK);
     } else {
         /* clear endpoint data toggle bit */
         USBD_RX_DTG_CLEAR(ep_num);
 
         udev->transc_out[ep_num].ep_stall = 0U;
 
         /* clear endpoint stall status */
-        USBD_EP_RX_STAT_SET(ep_num, EPRX_VALID);
+        USBD_EP_RX_STAT_SET(ep_num, EPRX_NAK);

```

Comment:
Is this a bug in the low level firmware that should eventually be fixed upstream?

Reply:
Probably. The existing behavior is wrong, because the peripheral's packet buffers don't get reset on a bus reset, so re-enabling the endpoint retransmits the offending packet.

Arguably, the fix is incorrect for an isochronous endpoint, because NAK isn't a valid status for one (only VALID and DISABLED are allowed for isochronous).

Reply:
(When you get a chance, could you please create a github issue about upstreaming the fix, just so we don't lose track?)

Reply:
Done! keyboardio/Kaleidoscope#1313